test-jest/api/activities.test.js

```diff
@@ -0,0 +1,281 @@
+          get: jest.fn(() => ({ authorization: null }))
+        })
+      }
+      const href = 'http://example.com/path/to/somewhere'
+      const activityId = '1'
+      const submissionId = '1'
+      const riverId = '1'
+      const daysFishedWithMandatoryRelease = 1
+      const daysFishedOther = 1
+      mockRequest.mockImplementationOnce(() => ({ id: '1', _links: { self: { href: href } }, activityId, riverId, daysFishedOther, daysFishedWithMandatoryRelease }))
+      const result = await activitiesApi.change(request, { submissionId, activityId, riverId, daysFishedWithMandatoryRelease, daysFishedOther })
+      expect(result.id).toEqual('/path/to/somewhere')
+      expect(result.activityId).toEqual(activityId)
+      expect(result.daysFishedOther).toEqual(daysFishedOther)
+      expect(result.daysFishedWithMandatoryRelease).toEqual(daysFishedWithMandatoryRelease)
+      expect(result.riverId).toEqual(riverId)
+    })
+
+    it('should display error if there is an error', async () => {
+      const activitiesApi = new ActivitiesApi()
+      const request = {
+        cache: () => ({
+          get: jest.fn(() => ({ authorization: null }))
+        })
+      }
+      const errors = 'errors'
+      const href = 'http://example.com/path/to/somewhere'
+      const activityId = '1'
+      const submissionId = '1'
+      const riverId = '1'
+      const daysFishedWithMandatoryRelease = 1
+      const daysFishedOther = 1
+      mockRequest.mockImplementationOnce(() => ({ id: 1, errors, _links: { self: { href: href } }, activityId, riverId, daysFishedOther, daysFishedWithMandatoryRelease }))
+      const result = await activitiesApi.change(request, { submissionId, activityId, riverId, daysFishedWithMandatoryRelease, daysFishedOther })
+      expect(result.id).toEqual(1)
+    })
+  })
+
+  describe('sort', () => {
+    it('if in alphabetical order should not change', async () => {
+      const activitiesApi = new ActivitiesApi()
+      const a = {
+        river: {
+          name: 'a test'
+        }
+      }
+      const b = {
+        river: {
+          name: 'b test'
+        }
+      }
+
+      const result = await activitiesApi.sort(a, b)
+      expect(result).toEqual(-1)
+    })
+
+    it('if not in alphabetical order should rearrange', async () => {
+      const activitiesApi = new ActivitiesApi()
+      const a = {
+        river: {
+          name: 'b test'
+        }
+      }
+      const b = {
+        river: {
+          name: 'a test'
+        }
+      }
+
+      const result = await activitiesApi.sort(a, b)
+      expect(result).toEqual(1)
+    })
+
+    it('if same river they should not be rearranged', async () => {
+      const activitiesApi = new ActivitiesApi()
+      const a = {
+        river: {
+          name: 'a test'
+        }
+      }
+      const b = {
+        river: {
+          name: 'a test'
+        }
+      }
+
+      const result = await activitiesApi.sort(a, b)
+      expect(result).toEqual(0)
+    })
+  })
```

Comment:
This would be a separate, similar test:
```
it('keeps data order the same if the same river has been fished several times', () => {
  const activitiesApi = new ActivitiesApi()
  const riverData = [{river: { name: 'Tweed', dateFished: '2021-07-22'}}, {river: { name: 'Tweed', dateFished: '2021-07-15'}}]
  const result = riverData.sort(activitiesApi.sort)
  expect(result[0].dateFished).toEqual(result[0].dateFished)
})
```